ui/src/tool-panel/location-track/location-track-infobox.tsx

```diff
@@ -482,8 +487,16 @@ const LocationTrackInfobox: React.FC<LocationTrackInfoboxProps> = ({
                                                             delegates.onStartSplitting({
                                                                 locationTrack: locationTrack,
                                                                 allowedSwitches:
-                                                                    splitInitializationParameters?.switches ||
-                                                                    [],
+                                                                    splitInitializationParameters?.switches.filter(
```

Comment:
Täälläkin alkaa muuten olemaan selkeästi enemmän sisennystä kuin koodia. Yksi helppo tapa järkeistää tuota on tehdä tuo onClickin funkkari itse html:n yläpuolella. Toinen voisi olla tehdä tämä koko buttons-osio omassa funkkarissaan pääosasta erillään.

Reply:
Sama huomio.

Suosisin itse tapaa, jossa onClick-funkkarit eivät sisältäisi (max paria riviä pidempää) funktiorunkoa lainkaan, vaan olisivat mielummin viittauksia muualla määritettyyn funkkariin. Näin leiskasta+toiminnallisuudesta saisi lähestulkoon aina helpommin selvää ja samalla napin funktiolle päätyisi (hopefully) kuvaava nimikin. Hifistelynä onClick-funkkarit voisi toki vielä parametrisoidakin, jolloin ei tarvittaisi edes komponenttikontekstin muuttujien arvoja (-> puhtaampi funkkari).